### DISCUSSION TITLE
Laravel Queue job is dispatched infinitely

### DISCUSSION THREAD

**Discussion by market-south** [laravel/framework#47606] [Q&A]
I am building a Laravel application that integrates with an ERP system using Graph QL to perform query and mutation operations.

The queue job is dispatched inside a for a loop. i.e., If the ERP has 100 customers, the Laravel applications loop through each customer and dispatch a queue job to update the customer's details into the Laravel database.

The integration operation is running smoothly. But my question is why Laravel keeps dispatching jobs infinitely?. Based on my research, can understand Laravel will queue the job if the queue fails to retry the job. But there is no errors in laravel.log or failed queue jobs table. I can confirm this by checking the Laravel db and integration logs.

Why does Laravel keep dispatching the jobs in finery, where technically there should be only 100 jobs dispatched in my above example scenario?

Even tried running `php artisan queue:work --tries=1` where the job should not be running more than once

Tried other following options:

1. Tried Batch request to have all the 100 jobs in one batch, multiple batches are getting queued
2. Tried Chain request where technically the chain stops if one of the jobs in it fails, but it is running smoothly and after the end, another chain of the same 100 jobs gets queued and goes on.
3. Tried using uniqueID function with the customer ID as a key where the job should not be queued for a specified time.

None of them stops the jobs from queued infinitely.

I am wondering if should I move the loop inside a job. i.e., dispatch one job and perform the data loop inside a job and use `use Illuminate\Contracts\Queue\ShouldBeUnique`

Any help would be appreciated!

**DISPATCH JOB**

```
foreach($customers as $customer)
{
   SyncCustomer::dispatch($customer);
} 
```

**HANDLE JOB**

```
<?php

namespace App\Jobs;

use App\Http\Controllers\Helper;
use Exception;
use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldBeUnique;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;
use Throwable;

class SyncCustomer implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;


   protected $customer;
    /**
     * Create a new job instance.
     *
     * @return void
     */
    public function __construct($customer)
    {
        $this->customer = $customer;
        $this->recentOrders = [];
    }


    public function uniqueId(): string
    {
        return $this->customer->id;
    }


    public function handle()
    {
      try 
       {
         // handle operation
       }
     catch(Exception $e)
      {
        logger($e->getMessage());
        throw new Exception($e);
      }
    }

    public function failed(Throwable $th)
    {
        logger($th->getMessage());
    }
}
```

**Comment by yuan8** [comment]
- for check your code
have you tried in sync mode?
maybe   errors not write on log file.
